File: pyvulkan/21 - Textures/finished/scene.py
from config import *

class Scene:


    def __init__(self):

        self.triangle_positions = []
        self.square_positions = []
        self.star_positions = []

        y = -1.0
        while y < 1.0:
            self.triangle_positions.append(
                np.array([-0.3, y, 0], dtype = np.float32)
            )
            self.square_positions.append(
                np.array([0.0, y, 0], dtype = np.float32)
            )
            self.star_positions.append(
                np.array([0.3, y, 0], dtype = np.float32)
            )
            y += 0.2

File: APIs/vulkan 2022/24 - Obj Models/Start/model/scene.cpp
#include "scene.h"

/**
* Scene constructor
*/
Scene::Scene() {

	float x = -0.3f;
	for (float z = -1.0f; z <= 1.0f; z += 0.2f) {
		for (float y = -1.0f; y <= 1.0f; y += 0.2f) {

			trianglePositions.push_back(glm::vec3(x, y, z));

		}
	}

	x = 0.0f;
	for (float z = -1.0f; z <= 1.0f; z += 0.2f) {
		for (float y = -1.0f; y < 1.0f; y += 0.2f) {

			squarePositions.push_back(glm::vec3(x, y, z));

		}
	}

	x = 0.3f;
	for (float z = -1.0f; z <= 1.0f; z += 0.2f) {
		for (float y = -1.0f; y < 1.0f; y += 0.2f) {

			starPositions.push_back(glm::vec3(x, y, z));

		}
	}

};